### PR TITLE
[Qt5|qmake] pass toolchain setup

### PR DESCRIPTION
**Pull request by Neumann-A**
one todo in #9881

### CODE REVIEW

File: ports/qt5-base/cmake/configure_qt.cmake [hunk 1]
```diff
@@ -29,6 +29,46 @@ function(configure_qt)
...
+        # QMake assumes everything is on PATH?
+        vcpkg_list(APPEND ${var} "${qmake_var}=${prog}")
+        find_program(${qmake_var} NAMES "${prog}")
+        cmake_path(COMPARE "${${qmake_var}}" EQUAL "${value}" correct_prog_on_path)
+        if(NOT correct_prog_on_path AND NOT "${value}" MATCHES "|:")
+            message(FATAL_ERROR "Detect path mismatch for '${qmake_var}'. '${value}' is not the same as '${${qmake_var}}'. Please correct your PATH!")
+        endif()
+        unset(${qmake_var})
+        unset(${qmake_var} CACHE)
+        set(${var} "${${var}}" PARENT_SCOPE) # Is this correct? Or is there a vcpkg_list command for that?
```
Comment (JavierMatosD) [inline ports/qt5-base/cmake/configure_qt.cmake:45]: This is correct :)
```suggestion
        set(${var} "${${var}}" PARENT_SCOPE)
```

File: scripts/cmake/vcpkg_configure_qmake.cmake [hunk 2]
```diff
@@ -13,6 +13,55 @@ function(vcpkg_configure_qmake)
...
+        # QMake assumes everything is on PATH?
+        vcpkg_list(APPEND ${var} "${qmake_var}=${prog}")
+        find_program(${qmake_var} NAMES "${prog}")
+        cmake_path(COMPARE "${${qmake_var}}" EQUAL "${value}" correct_prog_on_path)
+        if(NOT correct_prog_on_path AND NOT "${value}" MATCHES "|:")
+            message(FATAL_ERROR "Detect path mismatch for '${qmake_var}'. '${value}' is not the same as '${${qmake_var}}'. Please correct your PATH!")
+        endif()
+        unset(${qmake_var})
+        unset(${qmake_var} CACHE)
+        set(${var} "${${var}}" PARENT_SCOPE) # Is this correct? Or is there a vcpkg_list command for that?
```
Comment (JavierMatosD) [inline scripts/cmake/vcpkg_configure_qmake.cmake:30]:
```suggestion
        set(${var} "${${var}}" PARENT_SCOPE)
```

File: scripts/cmake/vcpkg_configure_qmake.cmake [hunk 3]
```diff
@@ -13,6 +13,55 @@ function(vcpkg_configure_qmake)
...
+        qmake_append_program(qmake_build_tools "QMAKE_LIB" "${VCPKG_DETECTED_CMAKE_AR}")
+        qmake_append_program(qmake_build_tools "QMAKE_LINK" "${VCPKG_DETECTED_CMAKE_LINKER}")
+    else()
+        qmake_append_program(qmake_build_tools "QMAKE_LINK" "${VCPKG_DETECTED_CMAKE_CXX_COMPILER}")
+        qmake_append_program(qmake_build_tools "QMAKE_LINK_SHLIB" "${VCPKG_DETECTED_CMAKE_CXX_COMPILER}")
+        qmake_append_program(qmake_build_tools "QMAKE_LINK_C" "${VCPKG_DETECTED_CMAKE_C_COMPILER}")
+        qmake_append_program(qmake_build_tools "QMAKE_LINK_C_SHLIB" "${VCPKG_DETECTED_CMAKE_C_COMPILER}")
+    endif()
+    set(qmake_comp_flags "")
+    # Note sure about these. VCPKG_QMAKE_OPTIONS offers a way to opt out of these. (earlier values being overwritten by later values; = set +=append *=append unique -=remove)
```
Comment (JavierMatosD) [inline scripts/cmake/vcpkg_configure_qmake.cmake:57]: What are you not sure about?

File: scripts/cmake/vcpkg_configure_qmake.cmake [hunk 4]
```diff
@@ -13,6 +13,55 @@ function(vcpkg_configure_qmake)
...
+    qmake_append_program(qmake_build_tools "QMAKE_CXX" "${VCPKG_DETECTED_CMAKE_CXX_COMPILER}")
+    qmake_append_program(qmake_build_tools "QMAKE_AR" "${VCPKG_DETECTED_CMAKE_AR}")
+    qmake_append_program(qmake_build_tools "QMAKE_RANLIB" "${VCPKG_DETECTED_CMAKE_RANLIB}")
+    qmake_append_program(qmake_build_tools "QMAKE_STRIP" "${VCPKG_DETECTED_CMAKE_STRIP}")
+    qmake_append_program(qmake_build_tools "QMAKE_NM" "${VCPKG_DETECTED_CMAKE_NM}")
+    qmake_append_program(qmake_build_tools "QMAKE_RC" "${VCPKG_DETECTED_CMAKE_RC_COMPILER}")
+    qmake_append_program(qmake_build_tools "QMAKE_MT" "${VCPKG_DETECTED_CMAKE_MT}")
+    if(NOT VCPKG_TARGET_IS_WINDOWS OR VCPKG_DETECTED_CMAKE_AR MATCHES "ar$")
+        # This is the reason why users should probably use a
+        # customized qt.conf with more domain knowledge.
```
Comment (JavierMatosD) [inline scripts/cmake/vcpkg_configure_qmake.cmake:44]:
```suggestion
```

File: scripts/cmake/vcpkg_configure_qmake.cmake (hunk 3 above)
Comment (JavierMatosD) [inline scripts/cmake/vcpkg_configure_qmake.cmake:57]:
```suggestion
```

**Review by JavierMatosD — DISMISSED** [review]

File: scripts/cmake/vcpkg_configure_qmake.cmake (hunk 3 above)
Comment (Neumann-A) [reply scripts/cmake/vcpkg_configure_qmake.cmake:57]: If those flags are all the relevant ones. qmake documentation is a bit .... incomplete I would say.

**Comment by Cheney-W** [comment]
@JavierMatosD Please review again, thank you!

**Pull request merged**